sw/device/tests/otbn_isa_test.c

```diff
@@ -0,0 +1,144 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/otbn_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * This test runs every instruction in OTBN's ISA and checks the result.
+ *
+ * It reuses the otbn's `smoke_test.s`,
+ * but with accesses the `RND` and `KEY_*` WSRs removed.
+ * The result in `w1` and `w2` are downstream of the `KEY_*`,
+ * loads and so are not checked.
+ *
+ * The zero register and registers derived from symbol position are also
+ * ignored, namely `x0`, `x1` and `x31`.
+ */
+OTTF_DEFINE_TEST_CONFIG();
+
+OTBN_DECLARE_APP_SYMBOLS(smoke_test);
+OTBN_DECLARE_SYMBOL_ADDR(smoke_test, gpr_state);
+OTBN_DECLARE_SYMBOL_ADDR(smoke_test, wdr_state);
+
+static const otbn_app_t kAppSmokeTest = OTBN_APP_T_INIT(smoke_test);
+static const otbn_addr_t kGprState = OTBN_ADDR_T_INIT(smoke_test, gpr_state);
+static const otbn_addr_t kWdrState = OTBN_ADDR_T_INIT(smoke_test, wdr_state);
+
+// These expected values are taken from
+// `hw/ip/otbn/dv/smoke/smoke_expected.txt`.
```

Comment:
I'd suggest expanding this comment to make it clear that we're talking about both GPRs and WDRs.